src/b-splines/b-splines.jl

```diff
@@ -98,7 +101,7 @@ function BSplineInterpolation(::Type{TWeights}, A::AbstractArray{Tel,N}, it::IT,
     else
         T = typeof(zero(TWeights) * first(A))
     end
-    BSplineInterpolation{T,N,typeof(A),IT,typeof(axs)}(A, fix_axis.(axs), it)
+    BSplineInterpolation{T,N}(A, fix_axis.(axs), it)
```

Comment:
Could you explain this?

Reply:
fix_axis below might change the input type.
The current version would convert ‘fix_axis.(axs)’ back to ‘typeof(axs)’.